crates/indexer/src/models/v2_objects.rs

```diff
@@ -0,0 +1,167 @@
+        }
+        let resource = MoveResource::from_write_resource(
+            write_resource,
+            0, // Placeholder, this isn't used anyway
+            txn_version,
+            0, // Placeholder, this isn't used anyway
+        );
+
+        match V2TokenResource::from_resource(
+            &type_str,
+            resource.data.as_ref().unwrap(),
+            txn_version,
+        )? {
+            V2TokenResource::ObjectCore(inner) => Ok(Some((
+                Self {
+                    transaction_version: txn_version,
+                    write_set_change_index,
+                    object_address: resource.address.clone(),
+                    owner_address: Some(standardize_address(inner.owner.as_str())),
+                    state_key_hash: resource.state_key_hash.clone(),
+                    guid_creation_num: Some(inner.guid_creation_num.clone()),
+                    allow_ungated_transfer: Some(inner.allow_ungated_transfer),
+                    is_deleted: false,
+                },
+                CurrentObject {
+                    object_address: resource.address,
+                    owner_address: Some(standardize_address(inner.owner.as_str())),
+                    state_key_hash: resource.state_key_hash,
+                    allow_ungated_transfer: Some(inner.allow_ungated_transfer),
+                    last_guid_creation_num: Some(inner.guid_creation_num),
+                    last_transaction_version: txn_version,
+                    is_deleted: false,
+                },
+            ))),
+        }
+    }
+
+    fn from_delete_resource(
+        delete_resource: &DeleteResource,
+        txn_version: i64,
+        write_set_change_index: i64,
+    ) -> anyhow::Result<Option<(Self, CurrentObject)>> {
+        if delete_resource.resource.to_string() == "0x1::object::ObjectCore" {
+            let resource = MoveResource::from_delete_resource(
+                delete_resource,
+                0, // Placeholder, this isn't used anyway
+                txn_version,
+                0, // Placeholder, this isn't used anyway
+            );
+            Ok(Some((
+                Self {
+                    transaction_version: txn_version,
+                    write_set_change_index,
+                    object_address: resource.address.clone(),
+                    owner_address: None,
+                    state_key_hash: resource.state_key_hash.clone(),
+                    guid_creation_num: None,
+                    allow_ungated_transfer: None,
+                    is_deleted: true,
+                },
+                CurrentObject {
+                    object_address: resource.address,
+                    owner_address: None,
+                    state_key_hash: resource.state_key_hash,
+                    allow_ungated_transfer: None,
+                    last_guid_creation_num: None,
+                    last_transaction_version: txn_version,
+                    is_deleted: false,
```

Comment:
I wonder how are deletes handled from the object side?

Reply:
ugh.